lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -7,13 +7,134 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
   alias LambdaEthereumConsensus.StateTransition.Misc
   alias LambdaEthereumConsensus.StateTransition.Mutators
   alias LambdaEthereumConsensus.StateTransition.Predicates
+  alias LambdaEthereumConsensus.Utils.BitVector
   alias SszTypes
   alias SszTypes.Attestation
   alias SszTypes.BeaconState
   alias SszTypes.ExecutionPayload
+  alias SszTypes.SyncAggregate
   alias SszTypes.Validator
   alias SszTypes.Withdrawal
 
+  @spec process_sync_aggregate(BeaconState.t(), SyncAggregate.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_sync_aggregate(
+        %BeaconState{
+          slot: slot,
+          current_sync_committee: current_sync_committee,
+          validators: validators
+        } = state,
+        %SyncAggregate{
+          sync_committee_bits: sync_committee_bits,
+          sync_committee_signature: sync_committee_signature
+        }
+      ) do
+    # Verify sync committee aggregate signature signing over the previous slot block root
+    committee_pubkeys = current_sync_committee.pubkeys
+    sync_committee_bits_as_num = sync_committee_bits |> :binary.decode_unsigned()
+
+    sync_committee_bits =
+      <<sync_committee_bits_as_num::unsigned-integer-little-size(bit_size(sync_committee_bits))>>
+
+    participant_pubkeys =
+      Enum.with_index(committee_pubkeys)
+      |> Enum.reduce([], fn {public_key, index}, participant_pubkeys ->
+        if BitVector.set?(sync_committee_bits, index) do
+          participant_pubkeys ++ [public_key]
+        else
+          participant_pubkeys
+        end
```

Comment:
Thank you.

I used `Enum.flat_map()` because doesn't transform

Reply:
It's more of a nitpick, but you could use a `filter` + `map` to avoid having an `if`. Like this:

```elixir
Enum.with_index(committee_pubkeys)
|> Enum.filter(fn {_, index} -> BitVector.set?(sync_committee_bits, index) end)
|> Enum.map(fn {public_key, _} -> public_key end)
```

Reply:
Thanks. Done